src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesGcm.cs

```diff
@@ -22,16 +24,82 @@ public AesGcm(ReadOnlySpan<byte> key)
             ImportKey(key);
         }
 
+        [Obsolete(Obsoletions.AesGcmTagConstructorMessage, DiagnosticId = Obsoletions.AesGcmTagConstructorDiagId, UrlFormat = Obsoletions.SharedUrlFormat)]
         public AesGcm(byte[] key)
+            : this(new ReadOnlySpan<byte>(key ?? throw new ArgumentNullException(nameof(key))))
         {
-            ThrowIfNotSupported();
+        }
 
-            ArgumentNullException.ThrowIfNull(key);
+        /// <summary>
```

Comment:
The documentation on the new members is slightly out of character for how we write XML docs in System.Security, however it uses the existing documentation for AesGcm as much as possible, and in that style.